src/ui/server/controllers/insurance/business/company-details/index.POST.test.ts

```diff
@@ -155,15 +257,50 @@ describe('controllers/insurance/business/companies-details', () => {
       });
     });
 
-    it('should display validation error for companies house when company house error but trading name radio is selected', async () => {
+    it('should display validation error for companies house when company house error but trading name radio is selected and no trading address', async () => {
+      req.body = {
+        companiesHouseNumber: '123456',
+        [TRADING_NAME]: 'true',
+      };
+
+      const submittedValues = {
+        [COMPANY_HOUSE.INPUT]: req.body[COMPANY_HOUSE.INPUT],
+        [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
+        [TRADING_ADDRESS]: sanitiseValue(req.body[TRADING_ADDRESS]),
+      };
+
+      const getCompaniesHouseResponse = jest.fn(() => Promise.resolve({ apiError: true }));
+      api.keystone.getCompaniesHouseInformation = getCompaniesHouseResponse;
+
+      await post(req, res);
+
+      const errorMessage = EXPORTER_BUSINESS_ERROR[COMPANY_HOUSE.INPUT].TECHNICAL_ISSUES;
+
+      let validationErrors = generateValidationErrors(COMPANY_HOUSE.INPUT, errorMessage, {});
+      validationErrors = generateValidationErrors(TRADING_ADDRESS, errorMessageTradingAddress, validationErrors);
+
+      expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
+        ...corePageVariables({
+          PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...pageVariables(mockApplication.referenceNumber),
+        validationErrors,
+        submittedValues,
+      });
+    });
+
+    it('should display validation error for companies house when company house error but trading name radio is selected and there is trading address', async () => {
```

Comment:
Aren't all these validation rules covered in `generateValidationErrors`? Just thinking the controller unit tests could be a simpler because it just consumes `generateValidationErrors` rather than defining the rules `generateValidationErrors` is responsible.

Could just one test in the controller:

```js
describe('when there are validation errors', () => {
  it('should render template with validation errors', () => {
```

Reply:
I wanted to test all the errors together though